openshift/openshift-ansible-contrib
# GCP: Refactor data disk(s) creation and setup to instance template v2

#### What does this PR do?
Make additional disks part of the instance templates. This is a pre-requirement for scaling up. Unfortunately, GCP doesn't provide native way how to attach empty disk to instance within instance template, so we are using workaround including empty 1gb disk image. Contributed by @bdurrow Original PR: #505 

#### How should this be manually tested?
Verify that node instances have attached 2 additional disks, and that those disks are properly configured:
```
$ lsblk 
NAME                                                                                         MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
sda                                                                                            8:0    0   25G  0 disk 
└─sda1                                                                                         8:1    0   25G  0 part /
sdb                                                                                            8:16   0   25G  0 disk 
└─sdb1                                                                                         8:17   0   25G  0 part 
  ├─docker--vol-docker--pool_tmeta                                                           253:0    0   28M  0 lvm  
  │ └─docker--vol-docker--pool                                                               253:2    0 23.8G  0 lvm  
  │   ├─docker-8:1-12595346-cb4a5b21dc08aff11945180d602327f0080e53835b1c4acd0ead0a983fb1c239 253:3    0    3G  0 dm   
  │   ├─docker-8:1-12595346-c1aad8d38a50bca1df990833b6e7ca01b40f7c4a07a2d11e23c6f0807fde1f15 253:5    0    3G  0 dm   
  │   ├─docker-8:1-12595346-94e5ba10e0fe75e99f68d7c332de56765472c2bda1efb778e28b3cb5eb577ecf 253:6    0    3G  0 dm   
  │   └─docker-8:1-12595346-7fe666e2d3e213391d72ad2638516dd1328d1df9c1479937dcf860e66fba4962 253:7    0    3G  0 dm   
  └─docker--vol-docker--pool_tdata                                                           253:1    0 23.8G  0 lvm  
    └─docker--vol-docker--pool                                                               253:2    0 23.8G  0 lvm  
      ├─docker-8:1-12595346-cb4a5b21dc08aff11945180d602327f0080e53835b1c4acd0ead0a983fb1c239 253:3    0    3G  0 dm   
      ├─docker-8:1-12595346-c1aad8d38a50bca1df990833b6e7ca01b40f7c4a07a2d11e23c6f0807fde1f15 253:5    0    3G  0 dm   
      ├─docker-8:1-12595346-94e5ba10e0fe75e99f68d7c332de56765472c2bda1efb778e28b3cb5eb577ecf 253:6    0    3G  0 dm   
      └─docker-8:1-12595346-7fe666e2d3e213391d72ad2638516dd1328d1df9c1479937dcf860e66fba4962 253:7    0    3G  0 dm   
sdc                                                                                            8:32   0   50G  0 disk /var/lib/origin/openshift.local.volumes
```
Master instances should have attached one additional disk for docker in case of containerized deployment.

#### Is there a relevant Issue open for this?
Only the PR #505 

#### Who would you like to review this?
cc: @bdurrow @e-minguez @cooktheryan  PTAL

Resolves: #454 #447
